framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,123 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", false, "Find the difference with the highest absolute value");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference
+    const Real difference = _furthest_from_zero ? std::abs(_a[qp] - _b[qp]) : _a[qp] - _b[qp];
```

Comment:
I have tested from 1 to 8 tasks, and the X and Z locations are either  0.47886751345948 or 0.52113248654052. The Y locations is the same. I expect this behavior since the BCs are in Y and the ICs are symmetric in X and Z.

I dug a little deeper and noticed that the `_max_difference` for each case is actually different, just less than the tolerance of the CSVDiff. This is true for both MPI tasks and threaded cases. I added some code that performs a fuzzy equality on `_max_difference` to within `4*double_epsilon` and nothing changes. I have to increase the tolerance on the fuzzy equality to 1e-12 to start making matches. Parallel computation requires an even larger tolerance.

Should I be concerned about this? Any ideas on how to resolve this issue? I could simply create a non-symmetric problem that would pass the parallel checks, but am not sure if anyone sees a larger issue at hand that should be resolved first.